drivers/sensor/ak8975/ak8975.h

```diff
@@ -23,13 +23,13 @@
 #define AK8975_MEASURE_TIME_US		9000
 #define AK8975_MICRO_GAUSS_PER_BIT	3000
 
-#ifdef DT_INST_0_INVENSENSE_MPU9150_BASE_ADDRESS
-#if DT_INST_0_ASAHI_KASEI_AK8975_BASE_ADDRESS != 0x0C
+#if DT_NODE_HAS_PROP(DT_INST(0, invensense_mpu9150), reg)
+#if DT_REG_ADDR(DT_INST(0, asahi_kasei_ak8975)) != 0x0C
 #error "I2C address must be 0x0C when AK8975 is part of a MPU9150 chip"
 #endif
 
-#ifdef DT_INST_0_INVENSENSE_MPU9150
-#define MPU9150_I2C_ADDR		DT_INST_0_INVENSENSE_MPU9150_BASE_ADDRESS
+#if DT_HAS_DRV_INST(0)
```

Comment:
This still seems wrong, or at least confusing.  `DRV_COMPAT` here is `asahi_kasei_ak8975` not `invensense_mpu9150`, so the change suggests it's looking at a different node now.

I think this should be:

```
#if DT_NODE_HAS_PROP(DT_INST(0, invensense_mpu9150), reg)
#if DT_REG_ADDR(DT_INST(0, asahi_kasei_ak8975)) != 0x0C
#error "I2C address must be 0x0C when AK8975 is part of a MPU9150 chip"
#endif

#define MPU9150_I2C_ADDR                DT_INST_REG_ADDR(0)

#define MPU9150_REG_BYPASS_CFG          0x37
...
```
i.e. remove the condition around defining `MPU9150_I2C_ADDR`.  It was never necessary anyway: the context requires that `invensense_mpu9150`  exist with a `reg` property and that instance 0 of asahi_kasei_ak8975 have a `reg` property with a matching value, so you can just use it.

Reply:
Ok, should be fixed up now.